### PR TITLE
Adding weight window example

### PR DESCRIPTION
**Pull request by shimwell**
This PR attempts to add an example of the new openmc CPP weight windows using openmc lib.

I've made a shielding bunker room and run the simulation for the same compute time (30 seconds on this laptop) with and without weight windows enabled to show the improved flux maps that can be achieved with weight windows.

Many thanks to @pshriwise for adding this feature.

I actually made this example for my own [neutronics-workshop](https://github.com/fusion-energy/neutronics-workshop) but thought it might be worth sharing a little wider as the parametric geometry is quite nice and we are in need of a VR WW example over here

ps it might be worth squashing this PR like we do on the openmc repo now

### CODE REVIEW

**Review by pshriwise — CHANGES_REQUESTED** [review]
One small comment here to start. Thank you for this contribution @shimwell!

**Comment by shimwell** [comment]
Thanks Patrick,

by the way I think it might be worth squashing this repo like we do for the main openmc repo. The ipython notebooks contain images and these change on each commit making the full commit history relatively large

**Comment by pshriwise** [comment]
> Thanks Patrick,
> 
> by the way I think it might be worth squashing this repo like we do for the main openmc repo. The ipython notebooks contain images and these change on each commit making the full commit history relatively large

Agreed! I'll enable that merge style here.

**Comment by shimwell** [comment]
> consistent

I'm keen to make the plots better by adding color bars and axis labels. I can do this with matplotlib commands here on the example

If [this PR](https://github.com/openmc-dev/openmc/pull/2629) goes in then I can make these plots with a single function call and reduce the amount of code in this notebook considerably while also getting colorbars, axis titles etc.

So can I come back to addressing this request after the fate of [this PR](https://github.com/openmc-dev/openmc/pull/2629) has been decided

File: shielded_room_weight_window.ipynb [hunk 1]
```diff
@@ -0,0 +1,875 @@
...
+   "source": [
+    "def plot_mesh_tally(statepoint_filename, image_filename):\n",
+    "\n",
+    "    with openmc.StatePoint(statepoint_filename) as sp:\n",
+    "        flux_tally = sp.get_tally(name=\"flux tally\")\n",
+    "\n",
+    "    mesh_extent = mesh.bounding_box.extent['xy']\n",
+    "\n",
+    "    # get a slice of mean values on the xy basis mid z axis\n",
+    "    flux_mean = flux_tally.get_reshaped_data(value='mean', expand_dims=True, squeeze=True)#[:, :, 1]\n",
```
[the remark this answers is not in the document]
Comment (shimwell) [reply shielded_room_weight_window.ipynb]: address with https://github.com/openmc-dev/openmc-notebooks/pull/14/commits/7595f7e269a927568c42b554fbd58ee232c26c83

**Comment by pshriwise** [comment]
Taking another look at these today: I think something that might improve the clarity on the effect of variance reduction in this problem is to move the source upward so there's a less direct path for particles to enter the hallway adjacent to the room. We used this notebook as a starting point in our course at UW this spring and I thought was a nice demonstration! Thanks @shimwell

https://github.com/pshriwise/ne506-sp-24/blob/main/openmc/shielded_room_weight_window.ipynb

**Comment by pshriwise** [comment]
Going to go ahead and merge this for the nice contribution that it is. I have some additions that simplify the geometry creation from our latest NEA workshop and highlight use of the `Polygon` surface composite. I'll submit those in a subsequent PR.

Thanks @shimwell!

**Comment by shimwell** [comment]
Thanks @pshriwise , tag me in the follow up PR and i can review

**Pull request merged**